### PR TITLE
Change scaling factor from E8M0 to E8M23

### PR DESCRIPTION
**Pull request by ksivaman**

### CODE REVIEW

**Comment by nouiz** [comment]
Can you add in the description why this PR?

File: transformer_engine/jax/fp8.py [hunk 1]
```diff
@@ -426,8 +424,7 @@ def update_fp8_metas(state: Collection) -> Collection:
 
     .. code-block:: python
 
-        exp = floor(log2(fp8_max / amax)) - margin
-        sf = round(power(2, abs(exp)))
+        sf = (fp8_max / amax) / (2 ^ margin)
         sf = sf if amax > 0.0, else original_scale
         sf = sf if isfinite(amax), else original_scale)
         updated_scale = 1/sf if exp < 0, else sf
```
Comment (ptrendx) [inline transformer_engine/jax/fp8.py:430]: This needs to be changed too

**Review by ptrendx — APPROVED** [review]

File: transformer_engine/pytorch/fp8.py [hunk 2]
```diff
@@ -538,12 +538,9 @@ def _default_sf_compute(
     margin: int,
 ) -> torch.Tensor:
     """Default function to convert amax to scaling factor."""
-    exp = torch.floor(torch.log2(fp8_max / amax)) - margin
```
Comment (jiecaoyu) [inline transformer_engine/pytorch/fp8.py:541]: Hi @ksivaman, thanks for making this change! I wonder whether this change requires other kernel or code changes? The reason why I am asking is that [this doc](https://docs.nvidia.com/deeplearning/transformer-engine/user-guide/api/common.html) mentions "The scaling factor should always be a power of 2 to not introduce numerical error during the conversion from FP8 to higher precision format."

**Pull request merged**